### PR TITLE
Create tags on push

### PR DESCRIPTION
**Pull request by njooma**
This PR will compile and push protos to a PR when that PR is approved. Then, once the PR is merged, it will tag the release and push it to buf.

### CODE REVIEW

**Review by edaniels — APPROVED** [review]
yah sure

**Comment by njooma** [comment]
OK @edaniels I've tested this locally using act and on my own fork. Things seems to be working

File: .github/workflows/compile_protos.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Compile Protos
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request_review:
+    branches: [ 'main' ]
+    types: [submitted]
```
Comment (edaniels) [inline .github/workflows/compile_protos.yml:10]: shouldnt it be on approved?

**Review by edaniels — APPROVED** [review]
LGTM but lets make compile-protos a required check after

**Comment by njooma** [comment]
Should we enable auto merge as well once the check passes?

**Pull request merged**